From c26e32927c15252236cfe989be6d02cd025f6eab Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Mon, 12 Aug 2024 18:05:13 +0530
Subject: [PATCH] treewide: resolve the compiler warnings

Fixed:
qcom_nandc - excess elements in struct initializer
qcom_scm - ISO C90 forbids mixed declarations and code

Change-Id: I68f4a09c067fc76971fb6efe94d9768329702898
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 drivers/firmware/qcom_scm.c       | 8 ++++----
 drivers/mtd/nand/raw/qcom_nandc.c | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index 6a03404bf3ac..35d3fc1614f3 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -2915,10 +2915,6 @@ int qcom_scm_sdi_disable(struct device *dev)
 {
 	int ret;
 	struct qcom_scm_res res;
-	ret = qcom_scm_clk_enable();
-	if (ret)
-		return ret;
-
 	struct qcom_scm_desc desc = {
 		.svc = QCOM_SCM_SVC_BOOT,
 		.cmd = SCM_CMD_TZ_CONFIG_HW_FOR_RAM_DUMP_ID,
@@ -2928,6 +2924,10 @@ int qcom_scm_sdi_disable(struct device *dev)
 		.owner = ARM_SMCCC_OWNER_SIP,
 	};
 
+	ret = qcom_scm_clk_enable();
+	if (ret)
+		return ret;
+
 	ret = qcom_scm_call(__scm->dev, &desc, &res);
 
 	qcom_scm_clk_disable();
diff --git a/drivers/mtd/nand/raw/qcom_nandc.c b/drivers/mtd/nand/raw/qcom_nandc.c
index dcd1ab53e1f3..4f07cd37da39 100644
--- a/drivers/mtd/nand/raw/qcom_nandc.c
+++ b/drivers/mtd/nand/raw/qcom_nandc.c
@@ -323,7 +323,7 @@ static const u32 qspi_training_block_64[] = {
 struct nand_flash_dev qspinand_flash_ids_2k[] = {
 	{"MX35UF4GE4AD-Z4I SPI NAND 1G 1.8V",
 			{ .id = {0xc2, 0xb7} },
-		SZ_2K, SZ_256, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512), 0},
+		SZ_2K, SZ_256, SZ_128K, 0, 2, 128, NAND_ECC_INFO(8, SZ_512)},
 	{NULL}
 };
 
-- 
2.34.1

